[docs] [ENG-8973] Update docs re submit profiles (#25142)
expo/expo: docs/pages/submit/eas-json.mdx

}
```

> When you select a build for submission, then a profile matching the build profile used for the selected build will be chosen. If it does not exist, it will default to `production` profile.

Reviewer: ```suggestion
When you select a build for submission, it chooses the profile that is used for the selected build. If the profile does not exist, it selects the default `production` profile.
```

Let's use active voice here for better readability.

> You can also use EAS CLI to pick up another `submit` profile by specifying it with a parameter. For example, `eas submit --platform iOS --profile submit-profile-name`.

Reviewer 2: ```suggestion
You can also use EAS CLI to pick up another `submit` profile by specifying it with a parameter. For example, `eas submit --platform ios --profile submit-profile-name`.
```
